src/components/IFrame.js

```diff
@@ -54,6 +57,54 @@ export default class IFrame extends React.Component<Props, State> {
     window.removeEventListener('message', this.handleMessage);
   }
 
+  /*
+   * Extract a string-representation of the supplied metadata
+   */
+  encodeMetadata = (metadata: Object): string => {
+    try {
+        return JSON.stringify(metadata);
+    } catch (e) {
+      console.error(e);
+      return ''
+    }
```

Comment:
i feel like we shouldn't try-catch this... better to fail loudly than quietly omit metadata but succeed